vm/src/vm.rs

```diff
@@ -1013,6 +1020,42 @@ impl VirtualMachine {
             })
     }
 
+    pub fn decode(
+        &self,
+        obj: PyObjectRef,
+        encoding: Option<PyStringRef>,
+        errors: Option<PyStringRef>,
+    ) -> PyResult {
+        let codecsmodule = self.import("_codecs", &[], 0)?;
+        let decode = self.get_attribute(codecsmodule, "decode")?;
+        let mut args = vec![
+            obj,
+            encoding.map_or_else(|| self.get_none(), |s| s.into_object()),
```

Comment:
I think it would be more clear to do `errors.map_or_else(|| self.get_none(), |s| s.into_object())` as well instead of optionally to add them.

Reply:
If I did it like that it would throw an error because `errors` is an optional argument; it doesn't check that it's `None`, only whether or not it's present.